### PR TITLE
DHFPROD-5597: Show search link when facets > 20

### PR DESCRIPTION
**Pull request by wooldridge**
### Description

Popover search link only shown for a facet when number of facet items is great than a limit (20).

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by bsrikan** [comment]
@wooldridge this is a 5.4 story and should go into feature/5.4-develop branch

File: marklogic-data-hub-central/ui/src/components/facet/facet.test.tsx [hunk 1]
```diff
@@ -97,4 +99,35 @@ describe("Facet component", () => {
     expect(document.querySelector('[data-testid=sales_region-toggle] i svg')).toHaveStyle('transform: rotate(180deg);');
   });
 
+  it("Search link shown when facet number greater than limit" , () => {
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/facet/facet.test.tsx:102]: I would recommend 3 things:
1. Create a default facetProps under assets/mock-data to be used by all the facet tests
2. Move the < 20 test into this one. facetValues is already mocked out and has 12 items. Can use the default props to verify for < 20 condition
3. Do facetValues.push to get it above 20, rerender with updated facetValues and test for > 20 conddition. 

Refer https://wiki.marklogic.com/display/ENGINEERING/Writing+Data+Hub+Tests#WritingDataHubTests-RTLTests for guidance.

**Review by bsrikan — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/facet/facet.test.tsx (hunk 1 above)
Comment (wooldridge) [reply marklogic-data-hub-central/ui/src/components/facet/facet.test.tsx:102]: @bsrikan for tests of this functionality, I'd prefer to be able to set the limit as a constant and then key off of that for the tests (to make it clear what the limit is and make it easy to change). I can rewrite to test facet values for LIMIT-1, rerender, and then test values for LIMIT.

Comment (bsrikan) [reply marklogic-data-hub-central/ui/src/components/facet/facet.test.tsx:102]: You can still test the way you want to. You are going to use the default props which has less than 20 values, which is how `expect(queryByLabelText('popover-search-label')).not.toBeInTheDocument();` is being tested now. Only difference with my recommendation is both conditions will be asserted in same test making the expected behavior more apparent.

```
    const { getByLabelText, rerender } = render(<Facet {...data.facetProps} />)
    // Search link not shown for facets < 20
    expect(queryByLabelText('popover-search-label')).not.toBeInTheDocument();
    
    // Search link shown for facets >= 20
    const LIMIT =  20;
    const f = {
      "name": "facetName",
      "count": 123,
      "value": "facetValue"
    };
    let facetVals: any = [];
    for (let i = 0; i < LIMIT; i++) {
      facetVals.push(f); //run time appending to facetValues in assets/mock-data/entity-table
    }
    rerender(<Facet {...data.facetProps} facetValues={facetVals}/>)
    expect(getByLabelText('popover-search-label')).toBeInTheDocument();
```

Comment (wooldridge) [reply marklogic-data-hub-central/ui/src/components/facet/facet.test.tsx]: @bsrikan, thanks! I refactored the tests using a new facetProps default object, much cleaner.

**Review by Sanjeevani19 — APPROVED** [review]

**Review by Sanjeevani19 — APPROVED** [review]
@wooldridge Any reason for making it as two commits, if not could you please make them into one ?

**Review by brucean52 — DISMISSED** [review]

**Review by bsrikan — APPROVED** [review]
Thanks Mike for refactoring the test. Looks good.

**Comment by bsrikan** [comment]
@Sanjeevani19 @timur-isangulov seems to be failed e2e for the intermittent issue reported in this bug:
https://project.marklogic.com/jira/browse/DHFPROD-5392

**Comment by SameeraPriyathamTadikonda** [comment]
Looks like we need to fix the e2e tests.

**Comment by xnikhil08** [comment]
> Looks like we need to fix the e2e tests that failed on this PR

@wooldridge @timur-isangulov @Sanjeevani19 - Pushed a fix for those failing tests. The popover search related code can be removed from one of the test as its already being verified in RTL integration test. 
I removed the other test completely because its supposed to be handled in an RTL integration test and created a separate JIRA ticket for the same.
[https://project.marklogic.com/jira/browse/DHFPROD-5958]

**Comment by rjrudin** [comment]
This is marked as FixVersion=5.4.0 in JIRA, but it's pointed at the develop branch, so one of those is wrong - which one?

**Comment by wooldridge** [comment]
@rjrudin thanks for noticing this, this should go on feature/5.4-develop. I will create a new PR.

**Pull request closed without merging**